docs/syntaxes.md

```diff
@@ -0,0 +1,64 @@
+# Syntaxes
+
+Melange supports two syntaxes: OCaml syntax (the default) and Reason syntax.
+Choice of syntax doesn't change the output of the compiler, but Reason syntax is
+designed to be more approachable to JavaScript/TypeScript developers, and it
+supports the use of JSX. Most tooling, such as the [OCaml Platform VS Code
```

Comment:
Should I mention the possibility of using MLX with OCaml syntax? Or is that too experimental at this point?